recipe/build.sh

```diff
@@ -158,3 +159,9 @@ if [[ ("$target_platform" != "linux-ppc64le") && \
   # https://forum.hdfgroup.org/t/hdf5-1-10-long-double-conversions-tests-failed-in-ppc64le/4077
   make check RUNPARALLEL="mpiexec -n 2"
 fi
+
+# test for hdf5 C++ exceptions
+$CXX $RECIPE_DIR/testhdf5exc.cpp -o testhdf5exc -L$PREFIX/lib -lhdf5_cpp -lhdf5 -I$PREFIX/include
+LD_LIBRARY_PATH=$PREFIX/lib DYLD_LIBRARY_PATH=$PREFIX/lib ./testhdf5exc
```

Comment:
```suggestion
if [[ "${CONDA_BUILD_CROSS_COMPILATION:-}" != "1" || "${CROSSCOMPILING_EMULATOR}" != "" ]]; then
    ./testhdf5exc
fi
```
do you really need the LD_LIBRARY_PATH?

Reply:
the tests CANNOT run on OSX (and later windows) for arm64 builds since we don't have emulators for those.

Reply:
@gillins and @hmaarrfk, as you may have seen, the test is failing to build under linux-64:
```
+ x86_64-conda-linux-gnu-c++ /home/conda/recipe_root/testhdf5exc.cpp -o testhdf5exc -L/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib -lhdf5_cpp -lhdf5 -I/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: warning: libcrypto.so.3, needed by /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so, not found (try using -rpath or -rpath-link)
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: warning: libcurl.so.4, needed by /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so, not found (try using -rpath or -rpath-link)
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: warning: libsz.so.2, needed by /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so, not found (try using -rpath or -rpath-link)
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: warning: libz.so.1, needed by /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so, not found (try using -rpath or -rpath-link)
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `curl_easy_perform'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `EVP_sha256@OPENSSL_3.0.0'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `curl_easy_init'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `curl_slist_append'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `compress2'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `compressBound@ZLIB_1.2.0'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `HMAC@OPENSSL_3.0.0'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `inflate'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `inflateInit_'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `SHA256@OPENSSL_3.0.0'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `curl_easy_cleanup'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `inflateEnd'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `curl_slist_free_all'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `SZ_encoder_enabled'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `curl_easy_setopt'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `curl_global_init'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `SZ_BufftoBuffDecompress'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `SZ_BufftoBuffCompress'
/home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/conda/feedstock_root/build_artifacts/hdf5_1729557442008/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libhdf5.so: undefined reference to `curl_global_cleanup'
collect2: error: ld returned 1 exit status

```

Reply:
It might make more sense to add the test to `run_tests.sh` instead of `build.sh` and to build it with `cmake` similarly to:
https://github.com/conda-forge/hdf5-feedstock/blob/main/recipe/run_test.sh#L37-L40
https://github.com/conda-forge/hdf5-feedstock/blob/main/recipe/test/CMakeLists.txt
https://github.com/conda-forge/hdf5-feedstock/blob/main/recipe/test/h5_cmprss.c
That's what I originally had in mind.  I think `cmake` will figure out linking in the necessary libs and it's not worth trying to duplicate that effort some other way.

Reply:
On the other hand, maybe it makes sense to take the test back out since it didn't fail in https://github.com/conda-forge/hdf5-feedstock/pull/234